app/middleware/activity.js

```diff
@@ -0,0 +1,78 @@
+const User = require('../models/User')
+const redis = require('../../config/redis')
+const activity = async (req, res, next) => {
+  var redisClient = redis.redisClient
+  var route = req.originalUrl.replace(/\?.*$/, '')
+  if (route === '/user/logout') {
+    // Fetch data from redis
+    const token = req.token
+    var routeNamesDict
+    var mailid
+    const data = await User.findOne({
+      'tokens.token': token
```

Comment:
use user-id instead of the token as by default indexing is done on _id in mongoose so it would give better performance.